src/NewTriggerDeliveryPolicyPage.tsx

```diff
@@ -0,0 +1,135 @@
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+/* eslint-disable camelcase */
+
+import React, { useCallback, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Button, Container, Row, Spinner } from 'react-bootstrap';
+
+import { AlertsBanner, useAlerts } from './AlertManager';
+import { useAstarte } from './AstarteManager';
+import BackButton from './ui/BackButton';
+import TriggerDeliveryPolicyEditor from './components/TriggerDeliveryPolicyEditor';
+import {
+  AstarteTriggerDeliveryPolicyDTO,
+  AstarteTriggerDeliveryPolicyHandlerDTO,
+} from './astarte-client/types/dto/triggerDeliveryPolicy';
+
+export default (): React.ReactElement => {
+  const [policyDraft, setPolicyDraft] = useState<AstarteTriggerDeliveryPolicyDTO>();
+  const [isValidPolicy, setIsValidPolicy] = useState(false);
+  const [isInstallingPolicy, setIsInstallingPolicy] = useState(false);
+  const [isSourceVisible, setIsSourceVisible] = useState(true);
+  const [installationAlerts, installationAlertsController] = useAlerts();
+  const astarte = useAstarte();
+  const navigate = useNavigate();
+  let newHandlers: AstarteTriggerDeliveryPolicyHandlerDTO[] = [];
+
+  const handleToggleSourceVisibility = useCallback(() => {
+    setIsSourceVisible((isVisible) => !isVisible);
+  }, []);
+
+  const handlePolicyChange = useCallback(
+    (updatedPolicy: AstarteTriggerDeliveryPolicyDTO, isValid: boolean) => {
+      setPolicyDraft(updatedPolicy);
+      setIsValidPolicy(isValid);
+    },
+    [],
+  );
+
+  const validatePolicy = (policy?: AstarteTriggerDeliveryPolicyDTO) => {
+    policy?.error_handlers.forEach((el, index) => {
+      const str = el.on.toString();
+      const array = Array.from(str.split(','), Number);
+      const newArray = array.filter((v, i) => array.indexOf(v) === i);
+      if (newArray.length) {
+        newHandlers = [...newHandlers, { on: newArray, strategy: el.strategy }];
+      } else {
+        newHandlers.push(el);
+      }
+    });
+  };
+
+  const parsedErrorMessage = (status: number): string => {
+    switch (status) {
+      case 400:
+        return 'Bad request';
+      case 401:
+        return 'Authorization information is missing or invalid.';
+      case 403:
+        return 'Authorization failed for the resource. This could also result from unexisting resources.';
+      case 409:
+        return 'A trigger delivery policy with this name already exists.';
+      case 422:
+        return 'The provided trigger delivery policy is not valid.';
+      default:
+        return 'Not found';
+    }
+  };
+
+  const handleInstallPolicy = useCallback(() => {
+    validatePolicy(policyDraft);
```

Comment:
validatePolicy isn't reused somewhere else in the code and it uses an external var (`newHandlers`) as its state. You can just define it inline here or make it a pure function and define it oustide the element definition.